### ISSUE TITLE
[BUG] The matchesSource condition from <javaclass> is not being taken into account

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#356]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

0.3.0-alpha.5

### Priority

Minor

### Current Behavior

The [matchesSource](https://access.redhat.com/documentation/en-us/migration_toolkit_for_applications/5.2/html-single/rules_development_guide/index#javaclass_element_attributes) attribute in the `javaclass` condition from windup enables an additional filter on a line matched by adding a regex that the source code must also match. This is currently not happening in konveyor, causing false positives to show up for (some rules)[https://github.com/search?q=repo%3Awindup%2Fwindup-rulesets%20matchesSource&type=code].

See konveyor matches for rule `local-storage-00002`:
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/e20515f0-c231-48b7-b0ff-44d9ff330d31)

And windup matches for the same application and the same rule:
![image](https://github.com/konveyor/analyzer-lsp/assets/4527008/4c724bc8-90aa-462e-99c8-d4a2c8921753)

### Expected Behavior

The java conditions should also have the possibility of filtering by source code to enable this additional pass to be made, or have some similar mechanism.

### How Reproducible

Always (Default)

### Steps To Reproduce

1. In the given environment, analyze the application.
2. Go to issues, and find the issue "File system - java.net.URL/URI".
3. Analyze the same application with windup (see the given windup environment) and compare with the konveyor result: `./windup-cli --target eap8 --target openjdk17 --target cloud-readiness --target quarkus --target camel --target azure-appservice --input ~/Dev/redhat/40SampleApps/sample.daytrader7 ~/Dev/redhat/40SampleApps/sample.daytrader7.dependencies --overwrite --analyzeKnownLibraries`.

### Environment

```markdown
* Fedora 38
* minikube version: v1.31.2
* Konveyor 0.3.0-alpha.5
* [sample.daytrader7 app (master branch)](https://github.com/WASdev/sample.daytrader7)
* Analysis configuration:
** Source code + dependencies
** eap8, quarkus, openjdk17, azure, cloud-readiness, camel3
** Application and internal dependencies only
** No custom rules
** No additional advanced options

Windup environment:
* Java 11
* Windup (latest version)
```

**Comment by shawn-hurley** [comment]
I feel like this has to be kicked out to the next release

**Comment by shawn-hurley** [comment]
More specifically as we don't have any ability to do this, IMO this would be a whole new feature. I worry about adding it this late in the game.

**Comment by jmle** [comment]
@shawn-hurley I'd say this is a minor one. Only two or three rules actually use it. What do you think @rromannissen ?

**Comment by shawn-hurley** [comment]
I actually wonder if you can chain the violation to a file text search for finding a specific regex. If this is three rules, I would think that we could test with one and if it works just use that for this. I think you may even be able to add this to the shim. 

Let me know if that makes sense, and/or if we want to chat about this.
